lib/apiservers/engine/backends/container.go

```diff
@@ -476,6 +473,36 @@ func (c *ContainerBackend) ContainerExecStart(ctx context.Context, eid string, s
 		op.Errorf("Failed to start Exec task for container(%s) due to error (%s)", id, err)
 		return err
 	}
+
+	// now that the exec has begun and the streams are attached we need to inspect and then wait.
```

Comment:
The operation function blocks on attach so we will not get here until the streams have closed after a successful exec if `attach == true` and I don't think we should be waiting if `attach == false`.

Reply:
This comment has been corrected.